src/components/CompanyInfo/CredentialsFormModal.tsx

```diff
@@ -0,0 +1,289 @@
+          password: formData.details_password as string
+        }
+
+        const requestBody: UpdateCredentialsRequest = {
+          name: formData.name as string,
+          description: formData.description as string,
+          type: formData.type as string,
+          details: details,
+          owner: owner.businessId
+        }
+
+        const httpRequest = mode === 'edit' && credential?.publicId !== undefined
+          ? HttpClient.put(`/api/v1/credentials/` + credential?.publicId, requestBody, getHeaders(tokenResult.accessToken))
+          : HttpClient.post(`/api/v1/credentials`, requestBody, getHeaders(tokenResult.accessToken))
+
+        httpRequest.then((_response) => {
+          updateCredentialsCallback(owner, tokenResult.accessToken)
+          modalStateCallback(false)
+        }, handleApiError)
+      },
+      (error) => {
+        // TODO: show alert
+        return Promise.reject(error)
+      }
+    )
+  }
+
+  const handleApiError = (error: any) => {
+    if (error.response.status === 409) {
+      setFormErrors({
+        credential: t('formValidation:exists')
+      })
+    } else if (error.response.status === 404) {
+      setFormErrors({
+        credential: t('formValidation:notExists')
+      })
+    } else {
+      setFormErrors({
+        credential: error.message as string
+      })
+    }
+    return Promise.reject(error)
+  }
+
+  const updateFormState = useCallback((newFormData: Map | null, newFormErrors: Map | null) => {
+    if (newFormData) {
+      console.log("new form data ", JSON.stringify(newFormData))
+      setFormData(newFormData)
+    }
+    if (newFormErrors) {
+      setFormErrors(newFormErrors)
+    }
+  }, [])
+
+  const useGeneralListener: EventListenerOrEventListenerObject = useCallback(
+    (e: Event) => {
+      generalListener(e, updateFormState, formData, formErrors)
+    },
+    [updateFormState, formData, formErrors]
+  )
+
+  useEffect(() => {
+    const credentialsElement = document.querySelector('[id="name"]')
+    if (credentialsElement && credentialsElement.getAttribute('listener') !== 'true') {
+      credentialsElement.addEventListener('change', useGeneralListener)
+    }
+    const credentialsDescriptionElement = document.querySelector('[id="description"]')
+    if (credentialsDescriptionElement && credentialsDescriptionElement.getAttribute('listener') !== 'true') {
+      credentialsDescriptionElement.addEventListener('change', useGeneralListener)
+    }
+    const credentialsTypeElement = document.querySelector('[id="type"]')
+    if (credentialsTypeElement && credentialsTypeElement.getAttribute('listener') !== 'true') {
+      credentialsTypeElement.addEventListener('select', useGeneralListener)
+    }
+    const credentialsDetailsElement = document.querySelector('[id="details"]')
+    if (credentialsDetailsElement && credentialsDetailsElement.getAttribute('listener') !== 'true') {
+      credentialsDetailsElement.addEventListener('change', useGeneralListener)
+    }
```

Comment:
Check if the form works without this + accompanying `<div ...>` wrap below, might be this doesn't do anything.